pkg/disks/block.go

```diff
@@ -0,0 +1,43 @@
+package disks
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+func GetBlockSize(device string, sysfsPath string) (int, error) {
+	stat := syscall.Stat_t{}
+	err := syscall.Stat(device, &stat)
+	if err != nil {
+		return 0, fmt.Errorf("can't stat device %q: %w", device, err)
+	}
+
+	major, minor := unix.Major(stat.Rdev), unix.Minor(stat.Rdev)
+	blockSizeRaw, err := os.ReadFile(path.Join(sysfsPath, fmt.Sprintf("/dev/block/%d:%d/queue/logical_block_size", major, minor)))
+	if err != nil {
+		return 0, fmt.Errorf("can't read device %d:%d logical block size file: %w", major, minor, err)
+	}
+
+	blockSize, err := strconv.Atoi(strings.TrimSpace(string(blockSizeRaw)))
+	if err != nil {
+		return 0, fmt.Errorf("can't parse block size %q: %w", string(blockSizeRaw), err)
+	}
+
+	// The minimum block size for crc enabled filesystems is 1024, and it also cannot be smaller than the logical block size.
+	blockSize = max(1024, blockSize)
+
+	return blockSize, nil
+}
+
+func max(a, b int) int {
```

Comment:
make it generic and move to separate package? Helpers, algorithms?

Reply:
moved to separate package